src/app/screens/Login/index.tsx

```diff
@@ -38,11 +43,11 @@ function Login() {
         value={password}
       />
       <Button
-        buttonStyle={{ ...styles.loginButton, ...(!formIsValid && styles.loginButtonDisabled) }}
+        buttonStyle={{ ...styles.loginButton, ...(submitDisabled && styles.loginButtonDisabled) }}
         buttonTextStyle={styles.loginButtonText}
         onPress={signIn}
-        disabled={!formIsValid}>
-        Sign in
+        disabled={submitDisabled}>
+        {userIsLoading ? 'Loading...' : 'Sign in'}
```

Comment:
Here you can use the [ActivityIndicator](https://reactnative.dev/docs/activityindicator) for a more dynamic loading animation.
For example, you can pass a prop (`loading`)  to the Button and then:
```
Button() {
...
 return loading ? (
  <ActivityIndicator ... />
  ) : (
  <TouchableOpacity ... />
 );
...
}
```
What do you think?

Reply:
Yeah! Pretty cool. This change has a card that I'm gonna do in the next PR.
https://trello.com/c/sl7sTZV6